batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/ArtifactFactoryService.java

```diff
@@ -141,19 +144,34 @@ private BeanManagerImpl getBeanManager() {
         return beanManager == null ? null : BeanManagerProxy.unwrap(beanManager);
     }
 
-    private static Bean<?> getBean(final String ref, final BeanManager beanManager) {
+    private static Bean<?> getBean(final String ref, final BeanManager beanManager, final ClassLoader classLoader) {
         if (beanManager == null) {
             return null;
         }
-        BatchLogger.LOGGER.tracef("Looking up bean reference for '%s'", ref);
-        final Set<Bean<?>> beans = beanManager.getBeans(ref);
-        final Bean<?> bean = beanManager.resolve(beans);
-        if (bean != null) {
-            BatchLogger.LOGGER.tracef("Found bean '%s' for reference '%s'", bean, ref);
-        } else {
-            BatchLogger.LOGGER.tracef("No bean found for reference '%s;'", ref);
+        //TODO call findBean method directly after fully switching to EE10
+        try {
+            final Method findBeanMethod = WildFlySecurityManager.doUnchecked(new PrivilegedExceptionAction<Method>() {
+                @Override
+                public Method run() throws Exception {
+                    final Method m = AbstractArtifactFactory.class.getDeclaredMethod("findBean", String.class, BeanManager.class, ClassLoader.class);
+                    m.setAccessible(true);
+                    return m;
+                }
+            });
+            final Object result = findBeanMethod.invoke(null, ref, beanManager, classLoader);
+            BatchLogger.LOGGER.tracef("Found bean: %s", result);
+            return (Bean<?>) result;
+        } catch (Exception e) {
+            BatchLogger.LOGGER.tracef("Looking up bean reference for '%s'", ref);
+            final Set<Bean<?>> beans = beanManager.getBeans(ref);
+            final Bean<?> bean = beanManager.resolve(beans);
+            if (bean != null) {
+                BatchLogger.LOGGER.tracef("Found bean '%s' for reference '%s'", bean, ref);
+            } else {
+                BatchLogger.LOGGER.tracef("No bean found for reference '%s;'", ref);
```

Comment:
NIT: Is the semicolon intended: '%s;' ?

Reply:
That was there before my change, but I'll take a note and clean them up next time.